Tests/MediumServiceTests.cs

```diff
@@ -0,0 +1,285 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoFixture;
+using Moq;
+using Moq.Protected;
+using NUnit.Framework;
+using PostMediumGitHubAction.Domain;
+using PostMediumGitHubAction.Services;
+
+namespace Tests;
+
+public class MediumServiceTests
+{
+    private readonly HttpResponseMessage _successfullCurrentMediumUserHttpResponse = new()
+    {
+        StatusCode = HttpStatusCode.OK,
+        Content = new StringContent(
+            "{\"data\":{\"id\":\"1840a7bacce6d851c032cfb7de25919c500506726fe203254bb43b629755919b5\",\"username\":\"some-username\",\"name\":\"some-name\",\"url\":\"https://medium.com/@philips\",\"imageUrl\":\"https://some-url.com\"}}")
+    };
+
+    private readonly HttpResponseMessage _successfullGetPublicationsFromMediumUserHttpResponse = new()
+    {
+        StatusCode = HttpStatusCode.OK,
+        Content = new StringContent(

```

Comment:
Consider extracting this string into a constant that is reused in different test cases.